CONTRIBUTING.txt

```diff
@@ -511,3 +511,121 @@ Bugs
 ----
 
 Please `report bugs on GitHub <https://github.com/scikit-image/scikit-image/issues>`_.
+
+Benchmarks
+----------
+
+While not mandatory for most pull requests, we ask that performance related
+PRs include a benchmark in order to clearly depict the use-case that is being
+optimized for. In this section we will review how to setup the benchmarks,
+and three commands ``asv dev``, ``asv run`` and ``asv continuous``.
+
+Prerequisites
+~~~~~~~~~~~~~
+Begin by installing `airspeed velocity <https://asv.readthedocs.io/en/stable/>`_
+in your development environment. Prior to installation, be sure to activate your
+development environment, then if using ``venv`` you may install the requirement with::
+
+  source skimage-dev/bin/activate
+  pip install asv
+
+If you are using conda, then the command::
+
+  conda activate skimage-dev
+  conda install asv
+
+is more appropriate. Once installed, it is useful to run the command::
+
+  asv machine
+
+To let airspeed velocity know more information about your machine.
+
+Writing a benchmark
+~~~~~~~~~~~~~~~~~~~
+To write  benchmark, add a file in the ``benchmarks`` directory which contains a
+a class with a ``setup`` method and a ``time_`` method.
+
+The setup method should contain all the code necessary to prepare your data
+before the function you wish to benchmark. The ``time_`` method
+should contain calls to the functions you wish to benchmark.
+
+For example, the ``setup`` method may include loading data into memory, and
+potentially the allocation of memory for results. We take for example the
+``TransformSuite`` benchmark:
+
+.. code-block:: python
+
+  import numpy as np
+  from skimage import transform
+
+  class TransformSuite:
+      """Benchmark for transform routines in scikit-image."""
+
+      def setup(self):
+          self.image = np.zeros((2000, 2000))
+          idx = np.arange(500, 1500)
+          self.image[idx[::-1], idx] = 255
+          self.image[idx, idx] = 255
+
+      def time_hough_line(self):
+          result1, result2, result3 = transform.hough_line(self.image)
+
+Here, the creation of the image is completed is in the ``setup`` method, and not
```

Comment:
I do not understand this sentence, I guess there is a typo.

Reply:
I just want to bring to the attention of the contributor that things like creating our loading the dataset should not happen in the time function.

Reply:
That's a good thought but may be redundant with my suggestion above. 
```suggestion
Here, the creation of the image is completed in the ``setup`` method, and not
```

Reply:
I prefer @lagru suggestion then. It's easier to understand for me.

Reply:
Then, it will be ready to be merged on my side.